### PR TITLE
Improve BenchmarkPageProcessor

### PR DESCRIPTION
**Pull request by yingsu00**
Previously BenchmarkPageProcessor filters out 100% rows, and this makes
project was not tested at all. This PR adds a new option for the
filter to pass all rows so that projection can be tested. It also moves the creation of data and processors into a new inner class BenchmarkData, which allows for easier future parameter based expansions.
 
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by sujay-jain — APPROVED** [review]
Looks good

**Comment by yingsu00** [comment]
Thanks @sujay-jain. @mbasmanova Do you want to take a look?

File: presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java [hunk 1]
```diff
@@ -136,13 +142,19 @@ private static Page createInputPage()
             return pageBuilder.build();
         }
 
-        // where shipdate >= '1994-01-01'
-        //    and shipdate < '1995-01-01'
-        //    and discount >= 0.05
-        //    and discount <= 0.07
-        //    and quantity < 24;
-        private static final RowExpression createFilterExpression(FunctionManager functionManager)
+
```
Comment (sachdevs) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java]: Unrelated line add?

**Review by sachdevs — APPROVED** [review]
LGTM

File: presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java [hunk 2]
```diff
@@ -108,6 +110,10 @@ public Page handCoded(BenchmarkData data)
         private Tpch1FilterAndProject handcodedProcessor;
         private Page inputPage;
 
+        @SuppressWarnings("unused")
+        @Param({"0.0", "1.0"})
```
Comment (mbasmanova) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java:137]: Current implementation can only support these two values, e.g. no filter and filter that never passes. Hence, it would be clearer to introduce a boolean parameter. 

```
@Param({"true", "false"})
private boolean filterAlwaysFails;
```

File: presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java [hunk 3]
```diff
@@ -274,6 +275,7 @@ public static void main(String[] args)
         Options options = new OptionsBuilder()
                 .verbosity(VerboseMode.NORMAL)
                 .include(".*" + BenchmarkPageProcessor.class.getSimpleName() + ".*")
+                .addProfiler(GCProfiler.class)
```
Comment (mbasmanova) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java:301]: What does this do?

Comment (yingsu00) [reply presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java:301]: > What does this do?

This will run the GC profiler. This is why you see the GC stats in the benchmark results:
```
Benchmark                                                         (projectionDataType)   Mode  Cnt       Score        Error   Units
BenchmarkPageProcessor.compiled                                                 BIGINT  thrpt   10    8374.449 ±   1023.859   ops/s
BenchmarkPageProcessor.compiled:·gc.alloc.rate                                  BIGINT  thrpt   10    1451.783 ±    179.079  MB/sec
BenchmarkPageProcessor.compiled:·gc.alloc.rate.norm                             BIGINT  thrpt   10  363640.395 ±      1.385    B/op
BenchmarkPageProcessor.compiled:·gc.churn.PS_Eden_Space                         BIGINT  thrpt   10    1448.711 ±   1904.852  MB/sec
BenchmarkPageProcessor.compiled:·gc.churn.PS_Eden_Space.norm                    BIGINT  thrpt   10  368448.852 ± 487748.838    B/op
BenchmarkPageProcessor.compiled:·gc.churn.PS_Survivor_Space                     BIGINT  thrpt   10       0.019 ±      0.054  MB/sec
BenchmarkPageProcessor.compiled:·gc.churn.PS_Survivor_Space.norm                BIGINT  thrpt   10       4.904 ±     13.838    B/op
BenchmarkPageProcessor.compiled:·gc.count                                       BIGINT  thrpt   10       6.000               counts
BenchmarkPageProcessor.compiled:·gc.time                                        BIGINT  thrpt   10      27.000                   ms
BenchmarkPageProcessor.compiled                                                 DOUBLE  thrpt   10    8805.494 ±    916.612   ops/s
BenchmarkPageProcessor.compiled:·gc.alloc.rate                                  DOUBLE  thrpt   10    1527.888 ±    159.617  MB/sec
BenchmarkPageProcessor.compiled:·gc.alloc.rate.norm                             DOUBLE  thrpt   10  363640.100 ±      0.011    B/op
BenchmarkPageProcessor.compiled:·gc.churn.PS_Eden_Space                         DOUBLE  thrpt   10    1716.294 ±   1812.190  MB/sec
BenchmarkPageProcessor.compiled:·gc.churn.PS_Eden_Space.norm                    DOUBLE  thrpt   10  414368.217 ± 436717.530    B/op
BenchmarkPageProcessor.compiled:·gc.churn.PS_Survivor_Space                     DOUBLE  thrpt   10       0.008 ±      0.023  MB/sec
BenchmarkPageProcessor.compiled:·gc.churn.PS_Survivor_Space.norm                DOUBLE  thrpt   10       1.942 ±      5.771    B/op
BenchmarkPageProcessor.compiled:·gc.count                                       DOUBLE  thrpt   10       7.000               counts
BenchmarkPageProcessor.compiled:·gc.time                                        DOUBLE  thrpt   10      30.000                   ms
```

File: presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java (hunk 2 above)
Comment (yingsu00) [reply presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java:137]: Changed the callers to the following:

```
        private final RowExpression createFilterExpression(FunctionManager functionManager)
        {
            if (!filterAlwaysFails) {
                return new ConstantExpression(true, BOOLEAN);
            }
            ...
```
and
```
            private boolean filter(int position, Block discountBlock, Block shipDateBlock, Block quantityBlock)
            {
                if (!filterAlwaysFails) {
                    return true;
                }
```

**Review by mbasmanova — APPROVED** [review]

**Comment by yingsu00** [comment]
@mbasmanova Thanks for reviewing! The tests are now green.
